problem_builder/step.py

```diff
@@ -251,8 +251,13 @@ def resource_string(path):
 
     def get_translation_content(self):
         try:
+            language = utils.translation.get_language().split('-')
+            if len(language) == 2:
+                new_lang = language[0] + "_" + language[1].upper()
+            else:
+                new_lang = language
             return self.resource_string('public/js/translations/{lang}/textjs.js'.format(
-                lang=utils.translation.get_language(),
```

Comment:
Didn't this work for Japanese? Why was it removed in favor of a manual approach?

Reply:
Here we are receiving language code like ja-jp, de-DE while we have repos ja_JP ,de_DE , etc. If we don't do like this the default en is returning. That is we need to do this.

Reply:
So for four character language code we need to change last 3 characters. `-` to `_` last two characters to upper case e.g. `jp` to `JP`.

Reply:
Might be good to have a comment here to make that more clear.

Reply:
@bradenmacdonald  Thank you for highlighting. I have added a comment. Please proceed.
Thanks